plugins/catalog-backend/api-report.md

```diff
@@ -315,6 +316,19 @@ export type DefaultCatalogCollatorFactoryOptions =
 // @public @deprecated (undocumented)
 export type DeferredEntity = DeferredEntity_2;
 
+// @public (undocumented)
+export function doEmit(
```

Comment:
Hmmmm I don't think we're ready to put this one here.

- Its name is very generic but it has a specific purpose (relations)
- Some awkward details of the original old API are exposed (compound refs instead of strings)
- What does even undefined targets mean? (we know from the code but seeing it right here makes it look a bit odd)
- The context is pretty explicit, if this interface were reinvented in a more purpose built way maybe we'd have something like `namespacePolicy: 'inherit-self'` or whatever but then we're quickly in api design land
- Which one of the relations is the incoming and outgoing one? Single object option with named arguments comes to mind

Overall, seeing that we might want to revamp the processor interface into something nicer down the line, it may not be worth doing this now. Could consider making the example into a multiline thing `const namespace = <compute the namespace>` and then two literal emits on the lines after each other instead perhaps. And not worry about list-shaped things in the example; it can be a single-valued field for simplicity and then the reader can figure out how to put an `if` or a `for` around them.

Does that make sense? Or would you prefer to go and redesign this one into an easy-to-read and purpose built thing? More like

```ts
export function createEmitHelper(emit: CatalogProcessorEmit_2, self: Entity) {
  emitRelationPair(options: {
    target: Array<CompoundEntityRef | string>;
    relationType: { incoming: string; outgoing: string; };
    namespacePolicy?: 'inherit-self' | 'require',
  }):void { /* ... */ }
};
```

or something like that. Which may actually be the shape of a future emit interface in the first place. But ... was it worth it? :)

Reply:
That's fair, I had a feeling this one might get push back. I've just found that the last half-dozen or so Adopters who added a custom Kind want into problems with relations and this `doEmit` function help get them sorted out. Also, was just trying to not have these large code examples in the docs that get easily outdated or broken. 

I'll think on this one and if I still have time after my other PRs come up with a refactor tomorrow 👍